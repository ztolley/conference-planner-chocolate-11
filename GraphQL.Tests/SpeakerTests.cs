using System;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using ConferencePlanner.GraphQL.Data;
using ConferencePlanner.GraphQL.Speakers;
using ConferencePlanner.GraphQL.Types;
using HotChocolate;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;

namespace GraphQL.Tests
{
    public class SpeakerTests
    {
        [Fact]
        public async Task Speaker_Schema_Changed()
        {
            // arrange
            // act
            ISchema schema = await new ServiceCollection()
                .AddPooledDbContextFactory<ApplicationDbContext>(
                    options => options.UseInMemoryDatabase("Data Source=conferences.db"))
                .AddGraphQL()
                .AddQueryType(d => d.Name("Query"))
                .AddType<SpeakerQueries>()
                .AddMutationType(d => d.Name("Mutation"))
                .AddType<SpeakerMutations>()
                .AddType<SessionType>()
                .AddType<SpeakerType>()
                .AddType<TrackType>()
                .EnableRelaySupport()
                .BuildSchemaAsync();
             
            // assert
            schema.Print().MatchSnapshot();
        }

        [Fact]
        public async Task AddSpeaker()
        {
            // arrange
            IServiceProvider services = new ServiceCollection()
                .AddPooledDbContextFactory<ApplicationDbContext>(
                    options => options.UseInMemoryDatabase("Data Source=conferences.db"))
                .AddGraphQL()
                .AddQueryType(d => d.Name("Query"))
                    .AddType<SpeakerQueries>()
                .AddMutationType(d => d.Name("Mutation"))
                    .AddType<SpeakerMutations>()
                .AddType<SessionType>()
                .AddType<SpeakerType>()
                .AddType<TrackType>()
                .EnableRelaySupport()
                .Services
                .BuildServiceProvider();

            // act
            IExecutionResult result = await services.ExecuteRequestAsync(
                QueryRequestBuilder.New()
                    .SetQuery(@"
                        mutation AddSpeaker {
                            addSpeaker(input: {
                                name: ""Dolly Smith""
                                bio: ""Folly is really cool""
                            webSite: ""http://dolly.com""
                        }) {
                            speaker {
                                id
                                    name
                                bio
                                    webSite
                            }
                        }
                    }")
                    .Create());

            // assert
            result.ToJson().MatchSnapshot();
        }
    }
}