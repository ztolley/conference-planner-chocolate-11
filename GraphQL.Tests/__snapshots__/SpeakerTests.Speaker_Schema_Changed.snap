schema {
  query: Query
  mutation: Mutation
}

"The node interface is implemented by entities that have a global unique identifier."
interface Node {
  id: ID!
}

type AddSpeakerPayload {
  speaker: Speaker
  errors: [UserError!]
}

type Attendee {
  id: Int!
  firstName: String!
  lastName: String!
  userName: String!
  emailAddress: String
  sessionsAttendees: [SessionAttendee!]!
}

type Mutation {
  addSpeaker(input: AddSpeakerInput!): AddSpeakerPayload!
}

type Query {
  node(id: ID!): Node
  speakers: [Speaker!]!
  speakerById(id: ID!): Speaker!
  speakersById(ids: [ID!]!): [Speaker!]!
}

type Session implements Node {
  id: ID!
  speakers: [Speaker]
  attendees: [Attendee]
  track: Track
  trackId: ID
  title: String!
  abstract: String
  startTime: DateTime
  endTime: DateTime
  duration: TimeSpan!
}

type SessionAttendee {
  sessionId: Int!
  session: Session
  attendeeId: Int!
  attendee: Attendee
}

type Speaker implements Node {
  id: ID!
  sessions: [Session]
  name: String!
  bio: String
  webSite: String
}

type Track implements Node {
  id: ID!
  sessions: [Session]
  name: String!
}

type UserError {
  message: String!
  code: String!
}

input AddSpeakerInput {
  name: String!
  bio: String
  webSite: String
}

"The `DateTime` scalar represents an ISO-8601 compliant date time type."
scalar DateTime

"The `ID` scalar type represents a unique identifier, often used to refetch an object or as key for a cache. The ID type appears in a JSON response as a String; however, it is not intended to be human-readable. When expected as an input type, any string (such as `\"4\"`) or integer (such as `4`) input value will be accepted as an ID."
scalar ID

"The `Int` scalar type represents non-fractional signed whole numeric values. Int can represent values between -(2^31) and 2^31 - 1."
scalar Int

"The `String` scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text."
scalar String

"The `TimeSpan` scalar represents an ISO-8601 compliant duration type."
scalar TimeSpan
